armi/utils/plotting.py

```diff
@@ -38,3 +53,724 @@ def colorGenerator(skippedColors=10):
+    """
+    Generate a plot showing the axial block and enrichment distributions of each assembly type in the core.
+
+    Parameters
+    ----------
+    core: Core
+        The core to plot assembly types of.
+
+    assems: list
+        list of assembly objects to be plotted.
+
+    plotNumber: integer
+        number of uniquely identify the assembly plot from others and to prevent plots from being overwritten.
+
+    maxAssems: integer
+        maximum number of assemblies to plot in the assems list.
+
+    showBlockAxMesh: bool
+        if true, the axial mesh information will be displayed on the right side of the assembly plot.
+    """
+
+    if assems is None:
+        assems = list(core.parent.blueprints.assemblies.values())
+    if not isinstance(assems, (list, set, tuple)):
+        assems = [assems]
+    if not isinstance(plotNumber, int):
+        raise TypeError("Plot number should be an integer")
+    if maxAssems is not None and not isinstance(maxAssems, int):
+        raise TypeError("Maximum assemblies should be an integer")
+
+    numAssems = len(assems)
+    if maxAssems is None:
+        maxAssems = numAssems
+
+    # Set assembly/block size constants
+    yBlockHeights = []
+    yBlockAxMesh = OrderedSet()
+    assemWidth = 5.0
+    assemSeparation = 0.3
+    xAssemLoc = 0.5
+    xAssemEndLoc = numAssems * (assemWidth + assemSeparation) + assemSeparation
+
+    # Setup figure
+    fig, ax = plt.subplots(figsize=(15, 15), dpi=300)
+    for index, assem in enumerate(assems):
+        isLastAssem = True if index == (numAssems - 1) else False
+        (xBlockLoc, yBlockHeights, yBlockAxMesh) = _plotBlocksInAssembly(
+            ax,
+            assem,
+            isLastAssem,
+            yBlockHeights,
+            yBlockAxMesh,
+            xAssemLoc,
+            xAssemEndLoc,
+            showBlockAxMesh,
+        )
+        xAxisLabel = re.sub(" ", "\n", assem.getType().upper())
+        ax.text(
+            xBlockLoc + assemWidth / 2.0,
+            -5,
+            xAxisLabel,
+            fontsize=13,
+            ha="center",
+            va="top",
+        )
+        xAssemLoc += assemWidth + assemSeparation
+
+    # Set up plot layout
+    ax.spines["right"].set_visible(False)
+    ax.spines["top"].set_visible(False)
+    ax.spines["bottom"].set_visible(False)
+    ax.yaxis.set_ticks_position("left")
+    yBlockHeights.insert(0, 0.0)
+    yBlockHeights.sort()
+    yBlockHeightDiffs = numpy.diff(
+        yBlockHeights
+    )  # Compute differential heights between each block
+    ax.set_yticks([0.0] + list(set(numpy.cumsum(yBlockHeightDiffs))))
+    ax.xaxis.set_visible(False)
+
+    ax.set_title("Assembly Designs for {}".format(core.name), y=1.03)
+    ax.set_ylabel("Thermally Expanded Axial Heights (cm)".upper(), labelpad=20)
+    ax.set_xlim([0.0, 0.5 + maxAssems * (assemWidth + assemSeparation)])
+
+    # Plot and save figure
+    ax.plot()
+    figName = core.name + "AssemblyTypes{}.png".format(plotNumber)
```

Comment:
Also probably not worth passing core in just for the name. Might make sense to pass name in on its own

Reply:
like I said. its the same code as before, but I can look into how many cases need to be changed

Reply:
fair enough